src/main/java/com/minecrafttas/tasmod/tickratechanger/TickrateChangerServer.java

```diff
@@ -165,8 +173,13 @@ public void advanceTick() {
      * Sends a {@link AdvanceTickratePacket} to all clients
      */
     private static void advanceClientTick() {
-    	TASmod.packetServer.sendToAll(new AdvanceTickratePacket());
-	}
+    	try {
+	    	// packet 8: advance tick on clients
+	    	TASmod.server.writeAll(ByteBuffer.allocate(4).putInt(8));
```

Comment:
Write something different than forcing everyone to do maths to allocate the correct amount...

Reply:
yeah I'll admit it, it might not've been the best solution. While it's definitely the fastest it's also supper annoying to use. Here's my solution that is both faster and less memory consuming:
- copy the tas recorders  buffer recycler and fill it with say 1000 buffers of 1kb each.
- then only send the data actually filled (only recently learned about absolute buffer positioning) and recycle the buffer.

As much as I know you LOVE the TAS Recorder, this one class is actually a pretty genius implementation of a thread safe recycler. Basically instead of allocating a new optimal buffer, we reuse existing buffers and note their position and capacity.

on a sidenote my github app crashed 4 times while writing this comment. Why does everyone hate tas recorder :(